banderwagon/element.go

```diff
@@ -362,7 +363,9 @@ func (p *Element) Neg(p1 *Element) *Element {
 }
 
 func (p *Element) ScalarMul(p1 *Element, scalar_mont *fr.Element) *Element {
-	p.inner.ScalarMul(&p1.inner, scalar_mont)
+	var bigScalar big.Int
+	scalar_mont.ToBigIntRegular(&bigScalar)
+	p.inner.ScalarMultiplication(&p1.inner, &bigScalar)
```

Comment:
This is the only part of the PR that creates some extra overhead. gnark-crypto switched from an API `ScalarMul` which took an scalar in Montgomery form, to `ScalarMultiplication` which takes the scalar as a big int.

So that means that we need to do `ToBigIntRegular` to do the proper transformation which:
- Has to do a `FromMont(...)` call, and
- A `SetBigInt(...)`.

Both things weren't necessary before.

_I think_ it shouldn't be a big deal, since it's called from two use cases:
1. When generating the precomputed points at startup... so maybe a little extra overhead, but nothing runtime related.
2. For "Q point rescaling" (IPA) in proof generation and verification. For safety in IPA, we do a `a * Q` where `a` is a `fr.Element` from the transcript. I don't think that is really a bottleneck at all considering the big picture of other generation/verification work.